SDK/AppCenter/Microsoft.AppCenter.WindowsDesktop/Utils/DefaultApplicationSettings.cs

```diff
@@ -6,33 +6,65 @@
 using System.Configuration;
 using System.IO;
 using System.Reflection;
+using System.Xml;
+using System.Xml.Linq;
 
 namespace Microsoft.AppCenter.Utils
 {
     public class DefaultApplicationSettings : IApplicationSettings
     {
         private const string FileName = "AppCenter.config";
+        private const string BackupFileName = "AppCenter.config.bak";
+        private const string CorruptedConfigurationWarning = "Configuration is corrupted. App Center could work incorrectly";
         private static readonly object configLock = new object();
         private static Configuration configuration;
 
         internal static string FilePath { get; private set; }
 
+        internal static string BackupFilePath { get; private set; }
+
         public DefaultApplicationSettings()
         {
             lock (configLock)
             {
-                configuration = OpenConfiguration();
+                try
+                {
+                    configuration = OpenConfiguration();
+                    CreateConfigurationFileBackup();
+                }
+                catch (Exception e)
```

Comment:
```suggestion
                  catch (Exception e) when (e is XmlException || e is ConfigurationErrorsException)
```
?

Reply:
And additional catch block smth like
```
 catch {
throw;
}
```
Should work, I think

Reply:
Fixed

Reply:
I do not see the fix

Reply:
Now there is only 'catch (Exception e)'